microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/opentracing/properties/MicroProfileOpenTracingProvider.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.opentracing.properties;
+
+import static com.redhat.microprofile.jdt.internal.opentracing.MicroProfileOpenTracingConstants.OPERATION_NAME_PROVIDER;
+import static com.redhat.microprofile.jdt.internal.opentracing.MicroProfileOpenTracingConstants.SKIP_PATTERN;
+import static com.redhat.microprofile.jdt.internal.opentracing.MicroProfileOpenTracingConstants.TRACED_ANNOTATION;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IJavaProject;
+
+import com.redhat.microprofile.jdt.core.AbstractStaticPropertiesProvider;
+import com.redhat.microprofile.jdt.core.IPropertiesCollector;
+import com.redhat.microprofile.jdt.core.SearchContext;
+import com.redhat.microprofile.jdt.core.utils.JDTTypeUtils;
+
+/**
+ * Properties provider that provides static MicroProfile Metrics properties
+ * 
+ * @author David Kwon
+ * 
+ * @see https://github.com/eclipse/microprofile-opentracing/blob/master/spec/src/main/asciidoc/configuration.asciidoc
+ *
+ */
+public class MicroProfileOpenTracingProvider extends AbstractStaticPropertiesProvider {
+
+	@Override
+	protected boolean isAdaptedFor(SearchContext context, IProgressMonitor monitor) {
+		IJavaProject javaProject = context.getJavaProject();
+		return (JDTTypeUtils.findType(javaProject, TRACED_ANNOTATION) != null);
+	}
+
+	@Override
+	protected void collectStaticProperties(SearchContext context, IProgressMonitor monitor) {
+		IPropertiesCollector collector = context.getCollector();
+		String docs = "Specifies a skip pattern to avoid tracing of selected REST endpoints.";
+		super.addItemMetadata(collector, SKIP_PATTERN, "java.util.Optional<java.lang.String>", docs, null, null,
```

Comment:
It seems type is `java.util.regex.Pattern` see https://github.com/eclipse/microprofile-opentracing/blob/master/spec/src/main/asciidoc/microprofile-opentracing.asciidoc#disabling-server-side-tracing. 

An idea (if it's easy) is to manage validation on MicroProfile LS side to check pattern is valid. (a naive idea is try to create a java.util.regex.Pattern with the property value by catching it and if exception is thrown we could display the error.